test/e2e/reconciler/partitionset/partitionset_test.go

```diff
@@ -193,7 +193,7 @@ func TestPartitionSet(t *testing.T) {
 		if len(partitions.Items) == 1 {
 			return true, ""
 		}
-		return false, fmt.Sprintf("expected 1 partition, but got %d", len(partitions.Items))
+		return false, fmt.Sprintf("expected 1 partition, but got %d, details: %s", len(partitions.Items), spew.Sdump(partitions.Items))
```

Comment:
I imagine this prints hundreds of lines into the log - any way to make it more minimal?

Reply:
Indeed. I tried first to only print out the last state outside of the loop but did not get what I expected. On the other hand it also provides the history.

Reply:
Sounds good - we can let it mature over time